util/certs/certs.go

```diff
@@ -70,7 +70,7 @@ func EncodePublicKeyPEM(key *rsa.PublicKey) ([]byte, error) {
 func DecodeCertPEM(encoded []byte) (*x509.Certificate, error) {
 	block, _ := pem.Decode(encoded)
 	if block == nil {
-		return nil, nil
+		return nil, errors.New("unable to decode PEM data")
```

Comment:
So in the last unit test _returns error if unable to generate client cert_, I ran into a panic because we were returning `nil` as an error here. 
I added this error along with some tests for it so as to get past that.

However, it seems to be a deliberate intent to not return an error here and I wanted to know if there was a specific reason why.
@vincepri @detiber any thoughts?